mlrun/runtimes/base.py

```diff
@@ -422,13 +422,19 @@ def run(
         )
         self._pre_run(run, execution)  # hook for runtime specific prep
 
+        # Verify run params and hyperparams
+        self.verify_run_params(run.spec.parameters)
+
         # create task generator (for child runs) from spec
-        task_generator = None
-        if not self._is_nested:
-            task_generator = get_generator(run.spec, execution)
+        task_generator = get_generator(run.spec, execution)
+        if task_generator:
+            # verify valid task parameters
+            tasks = task_generator.generate(run)
+            for task in tasks:
+                self.verify_run_params(task.spec.parameters)
 
         last_err = None
-        if task_generator:
+        if task_generator and not self._is_nested:
```

Comment:
what does `_is_nested` means? would be best to add a bit of bread crumbs

Reply:
iirc, is_nested means a child run